packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -211,41 +214,115 @@ pub(crate) async fn register_indexer_assets(
     let mut assets: Vec<IndexerAsset> = Vec::new();
 
     if let Some(mut multipart) = multipart {
+        let mut replace_indexer: bool = false;
+        let mut remove_data: bool = false;
+        let mut asset_bytes: Vec<(IndexerAssetType, hyper::body::Bytes)> = Vec::new();
+
+        while let Ok(Some(field)) = multipart.next_field().await {
+            let name = field.name().unwrap_or("").to_string();
+            let data = field.bytes().await.unwrap_or_default();
+            match name.as_str() {
+                "replace_indexer" => {
+                    replace_indexer = std::str::from_utf8(&data.to_owned())
+                        .map_err(|e| ApiError::OtherError(e.to_string()))?
+                        .parse::<bool>()
+                        .map_err(|e| ApiError::OtherError(e.to_string()))?;
+                    continue;
+                }
+                "remove_data" => {
+                    remove_data = std::str::from_utf8(&data.to_owned())
+                        .map_err(|e| ApiError::OtherError(e.to_string()))?
+                        .parse::<bool>()
+                        .map_err(|e| ApiError::OtherError(e.to_string()))?;
+                    continue;
+                }
+                name => {
+                    let asset_type =
+                        IndexerAssetType::from_str(name).expect("Invalid asset type.");
+                    asset_bytes.push((asset_type, data))
+                }
+            };
+        }
+
         queries::start_transaction(&mut conn).await?;
 
-        let indexer_exists = queries::get_indexer_id(&mut conn, &namespace, &identifier)
-            .await
-            .is_ok();
-        if indexer_exists {
-            if !config.replace_indexer {
+        let indexer_id =
+            queries::get_indexer_id(&mut conn, &namespace, &identifier).await;
+
+        // Check that the schema has not changed.
+        for (asset_type, data) in asset_bytes.iter() {
+            if *asset_type == IndexerAssetType::Schema {
+                // If the indexer already exists
+                if let Ok(indexer_id) = indexer_id {
+                    // The schema must be the same. This query returns an asset
+                    // if the bytes match. If it returns None (and the indexer
+                    // exists), it means that its schema is different.
+                    if let None = queries::asset_already_exists(
```

Comment:
@lostman Everything seems to work but this.

Repro steps:
- `forc index new`
- `forc index deploy`
- Add a new `type` to your schema
- Now `forc index deploy` again -- should fail due to mismatched schemas since you didn't include `--remove-data`
- For me, it seems to just pick up from where the last indexer started